x-pack/plugins/infra/public/components/log_stream/index.tsx

```diff
@@ -101,13 +114,23 @@ Read more at https://github.com/elastic/kibana/blob/master/src/plugins/kibana_re
         scale="medium"
         wrap={false}
         isReloading={isReloading}
-        isLoadingMore={false}
-        hasMoreBeforeStart={false}
-        hasMoreAfterEnd={false}
+        isLoadingMore={isLoadingMore}
+        hasMoreBeforeStart={hasMoreBefore}
+        hasMoreAfterEnd={hasMoreAfter}
         isStreaming={false}
         lastLoadedTime={null}
         jumpToTarget={noop}
-        reportVisibleInterval={noop}
+        reportVisibleInterval={({ fromScroll, pagesBeforeStart, pagesAfterEnd }) => {
+          if (!fromScroll) {
+            return;
+          }
+
+          if (pagesBeforeStart < PAGE_THRESHOLD) {
```

Comment:
It feels like we need to guard against repeated, rapid firing of `fetch*Entries()` here. If I slowly scroll to the beginning or end of the stream an avalanche of queries is fired. Could `loadingState` and `pageLoadingState` be used to avoid that?

Reply:
Yeah that should solve it, at least while the request is loading.